com.civicrm.hrjobroles/js/src/job-roles/controllers/job-roles.controller.js

```diff
@@ -1122,10 +1122,16 @@ define([
       }
 
       updatedRole = angular.copy(vm.editData[roleId]);
-      updatedRole.location = (updatedRole.location === undefined) ? updatedRole.location = '' : updatedRole.location;
-      updatedRole.level = (updatedRole.level === undefined) ? updatedRole.level = '' : updatedRole.level;
-      updatedRole.department = (updatedRole.department === undefined) ? updatedRole.department = '' : updatedRole.department;
-      updatedRole.region = (updatedRole.region === undefined) ? updatedRole.region = '' : updatedRole.region;
+      // Ensure location, level, department, region exist and not disabled
+      updateLocation = _.includes(_.keys(vm.LocationsData), updatedRole.location);
+      updateLevel = _.includes(_.keys(vm.LevelsData), updatedRole.level);
+      updateDepartment = _.includes(_.keys(vm.DepartmentsData), updatedRole.department);
+      updateRegion = _.includes(_.keys(vm.RegionsData), updatedRole.region);
+
+      updatedRole.location = (updatedRole.location === undefined || !updateLocation) ? updatedRole.location = '' : updatedRole.location;
```

Comment:
Please move the code to a separate function with a meaningful name, there is too much duplication.